database/build.go

```diff
@@ -27,6 +27,8 @@ const (
 	maxTitleLength = 1000
 	// Maximum message field length.
 	maxMessageLength = 2000
+	// Maximum error field length.
+	maxErrorLength = 500
```

Comment:
I agree with @kneal 👍 

We should explore expanding the number of chars we allow for the `error` field across all resources.

Doubling it from `500` -> `1000` sounds like a fair place to start 😃

Reply:
it's currently a somewhat rare occurrence (afaik) so handling the outliers this way instead of having the outliers shape the data size seems fine to me. do we expect this be more commonplace?

Reply:
My personal view would be I don't think that field has ever changed in size. So, maybe this is showing errors that could be growing in size. I also don't think `1000` is that big of a concern and then would save us having to parse errors and hopefully not cut off the important part. If we had to go beyond `1000` then I think truncating could be useful.

I'm also not sure if we use error wrapping everywhere in the worker/runtime to make this errors longer so depending how errors are handled or want to be handled long term in that codebase `1000` might be the norm but I honestly don't know. The `error:` field in the build resource is entirely for admins because thats where we populate a platform error. So, cutting off the data really only hurts ourselves for richer research. 

> We should explore expanding the number of chars we allow for the error field across all resources.

@jbrockopp probably not a bad idea since I don't think that's ever been done. I assume most of these char lengths are there original defaults.

Reply:
So if we bump the limit to 1000, do we still want to crop the value if it exceeds that limit? Or do we have another idea for an approach to that situation?

Reply:
The only other idea I have would be compression if we wanna keep the field small but keep the full error messages.

Reply:
Would you support compressing the other fields currently being processed by the `crop` method? (`message` and `title`)

Reply:
I don't know the answer to that one. I think for error it might make sense to crop because that field could be quite variable depending on how the error wrapping works. 

Title and Message might make sense to crop because those come from the SCM system attached and I would assume GitHub has limits on those. So, it might make sense to just mirror those limits.

Reply:
@ecrupper 

> So if we bump the limit to 1000, do we still want to crop the value if it exceeds that limit? Or do we have another idea for an approach to that situation?

I'm onboard with the idea of a "simple" cropping strategy similar to `title` and `message`.

In the future, if this becomes a problem with the `1000` char limit, then we introduce compression.

cc: @kneal @wass3r